src/libcrun/utils.h

```diff
@@ -414,4 +414,39 @@ get_process_exit_status (int status)
 uid_t get_overflow_uid (void);
 gid_t get_overflow_gid (void);
 
+/* Adapted from systemd.  Include space for the NUL byte.  */
+#define DECIMAL_STR_MAX(type)                                        \
+  ((size_t) 2U + (sizeof (type) <= 1 ? 3U : sizeof (type) <= 2 ? 5U  \
+                                        : sizeof (type) <= 4   ? 10U \
+                                        : sizeof (type) <= 8   ? 20U \
+                                                               : sizeof (int[-2 * (sizeof (type) > 8)])))
+
+#define _MAX(a, b) ((a > b) ? (a) : (b))
+
+/* A buffer long enough to hold either /proc/self/fd/$FD or a /proc/$PID/fd/$FD path.  */
+typedef char proc_fd_path_t[sizeof ("/proc/") + _MAX (sizeof ("self"), DECIMAL_STR_MAX (pid_t)) + sizeof ("/fd/") + DECIMAL_STR_MAX (int) + 1];
```

Comment:
is there any case where `sizeof ("self")` is larger than `DECIMAL_STR_MAX (pid_t)`? or is this just for documenting the possible values?

Reply:
no there is no such case, it is only for documenting the possible values since anyway it has no runtime cost (all these values are known at build time).  But if it doesn't really help, I can drop the sizeof("/proc/") part and simplify the code.  And I should be using `(sizeof ("self") - 1)` since sizeof includes `\0`

Reply:
:woman_shrugging: I figured it was for documentation. I think it makes sense as-is

Reply:
is the new version clearer?

Reply:
if it were me, I would unconditionally set to `DECIMAL_STR_MAX...` and document in a comment how that will always be a longer string than `self` that would reduce the code complexity quite a bit, and you're already documenting that fact. not a blocker though for me though, it is definitely clear now

Reply:
dropped the extra code and just left a comment